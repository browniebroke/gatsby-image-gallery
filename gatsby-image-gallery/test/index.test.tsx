import * as React from 'react'
import Gallery from '../src/index'
import { createRenderer } from 'react-test-renderer/shallow'
import { Layout } from 'gatsby-plugin-image'

const constrainedLayout: Layout = 'constrained'
const gatsbyImageShapeMock = (path: string) => {
  return {
    layout: constrainedLayout,
    width: 100,
    height: 100,
    backgroundColor: 'red',
    images: {
      sources: [
        {
          media: path,
          type: 'JPG',
          srcSet: path,
        },
      ],
      fallback: {
        src: path,
      },
      alt: path,
    },
  }
}

const unifiedImageShapeMock = (path: string, alt?: string) => ({
  thumb: gatsbyImageShapeMock(`/thumb${path}`),
  full: gatsbyImageShapeMock(path),
  thumbAlt: alt || '',
})

describe('Gallery component', () => {
  test('that it renders with empty props', () => {
    const renderer = createRenderer()
    renderer.render(<Gallery images={[]} />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })

  describe('Unified image prop', () => {
    test('that it renders image prop when no alt property is used', () => {
      const renderer = createRenderer()
      renderer.render(
        <Gallery
          images={[
            unifiedImageShapeMock('/images/image001.jpg'),
            unifiedImageShapeMock('/images/image002.jpg'),
            unifiedImageShapeMock('/images/image003.jpg'),
            unifiedImageShapeMock('/images/image004.jpg'),
            unifiedImageShapeMock('/images/image005.jpg'),
          ]}
        />
      )
      const result = renderer.getRenderOutput()
      expect(result).toMatchSnapshot()
    })

    test('that it renders image prop when alt property is used', () => {
      const renderer = createRenderer()
      renderer.render(
        <Gallery
          images={[
            unifiedImageShapeMock('/images/image001.jpg', '001'),
            unifiedImageShapeMock('/images/image002.jpg', '002'),
            unifiedImageShapeMock('/images/image003.jpg', '003'),
            unifiedImageShapeMock('/images/image004.jpg', '004'),
            unifiedImageShapeMock('/images/image005.jpg', '005'),
          ]}
        />
      )
      const result = renderer.getRenderOutput()
      expect(result).toMatchSnapshot()
    })

    test('that it renders image prop when only in some cases the alt property is used', () => {
      const renderer = createRenderer()
      renderer.render(
        <Gallery
          images={[
            unifiedImageShapeMock('/images/image001.jpg', '001'),
            unifiedImageShapeMock('/images/image002.jpg', '002'),
            unifiedImageShapeMock('/images/image003.jpg'),
            unifiedImageShapeMock('/images/image004.jpg'),
            unifiedImageShapeMock('/images/image005.jpg', '005'),
          ]}
        />
      )
      const result = renderer.getRenderOutput()
      expect(result).toMatchSnapshot()
    })
  })

  describe('lightboxOptions props', () => {
    const lightboxOptions = {
      imageLoadErrorMessage: 'Impossible de charger cette image',
      nextLabel: 'Image suivante',
      prevLabel: 'Image prÃ©cÃ©dente',
      zoomInLabel: 'Zoomer',
      zoomOutLabel: 'DÃ©zoomer',
      closeLabel: 'Fermer',
    }

    test('that the Lightbox options are accepted', () => {
      const renderer = createRenderer()
      renderer.render(
        <Gallery
          images={[
            unifiedImageShapeMock('/images/image001.jpg'),
            unifiedImageShapeMock('/images/image002.jpg'),
            unifiedImageShapeMock('/images/image003.jpg'),
          ]}
          lightboxOptions={lightboxOptions}
        />
      )

      const result = renderer.getRenderOutput()
      expect(result).toMatchSnapshot()
    })
  })

  describe('rowMargin prop', () => {
    const rowMargin = 0

    test('that the rowMargin prop is accepted', () => {
      const renderer = createRenderer()
      renderer.render(
        <Gallery
          images={[
            unifiedImageShapeMock('/images/image001.jpg'),
            unifiedImageShapeMock('/images/image002.jpg'),
            unifiedImageShapeMock('/images/image003.jpg'),
          ]}
          rowMargin={rowMargin}
        />
      )

      const result = renderer.getRenderOutput()
      expect(result).toMatchSnapshot()
    })
  })

  describe('onClose prop', () => {
    const onClose = () => {
      console.log('test onClose')
    }

    test('onClose prop is accepted', () => {
      const renderer = createRenderer()
      renderer.render(
        <Gallery
          images={[
            unifiedImageShapeMock('/images/image001.jpg'),
            unifiedImageShapeMock('/images/image002.jpg'),
            unifiedImageShapeMock('/images/image003.jpg'),
          ]}
          onClose={onClose}
        />
      )

      const result = renderer.getRenderOutput()
      expect(result).toMatchSnapshot()
    })
  })
})
