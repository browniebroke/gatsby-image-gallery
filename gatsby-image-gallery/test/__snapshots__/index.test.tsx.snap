// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component Unified image prop that it renders image prop when alt property is used 1`] = `
<React.Fragment>
  <styled.div
    margin={-15}
  >
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt="001"
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image001.jpg",
              "fallback": Object {
                "src": "/thumb/images/image001.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image001.jpg",
                  "srcSet": "/thumb/images/image001.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt="002"
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image002.jpg",
              "fallback": Object {
                "src": "/thumb/images/image002.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image002.jpg",
                  "srcSet": "/thumb/images/image002.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt="003"
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image003.jpg",
              "fallback": Object {
                "src": "/thumb/images/image003.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image003.jpg",
                  "srcSet": "/thumb/images/image003.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt="004"
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image004.jpg",
              "fallback": Object {
                "src": "/thumb/images/image004.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image004.jpg",
                  "srcSet": "/thumb/images/image004.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt="005"
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image005.jpg",
              "fallback": Object {
                "src": "/thumb/images/image005.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image005.jpg",
                  "srcSet": "/thumb/images/image005.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
  </styled.div>
</React.Fragment>
`;

exports[`Gallery component Unified image prop that it renders image prop when no alt property is used 1`] = `
<React.Fragment>
  <styled.div
    margin={-15}
  >
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image001.jpg",
              "fallback": Object {
                "src": "/thumb/images/image001.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image001.jpg",
                  "srcSet": "/thumb/images/image001.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image002.jpg",
              "fallback": Object {
                "src": "/thumb/images/image002.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image002.jpg",
                  "srcSet": "/thumb/images/image002.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image003.jpg",
              "fallback": Object {
                "src": "/thumb/images/image003.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image003.jpg",
                  "srcSet": "/thumb/images/image003.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image004.jpg",
              "fallback": Object {
                "src": "/thumb/images/image004.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image004.jpg",
                  "srcSet": "/thumb/images/image004.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image005.jpg",
              "fallback": Object {
                "src": "/thumb/images/image005.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image005.jpg",
                  "srcSet": "/thumb/images/image005.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
  </styled.div>
</React.Fragment>
`;

exports[`Gallery component Unified image prop that it renders image prop when only in some cases the alt property is used 1`] = `
<React.Fragment>
  <styled.div
    margin={-15}
  >
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt="001"
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image001.jpg",
              "fallback": Object {
                "src": "/thumb/images/image001.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image001.jpg",
                  "srcSet": "/thumb/images/image001.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt="002"
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image002.jpg",
              "fallback": Object {
                "src": "/thumb/images/image002.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image002.jpg",
                  "srcSet": "/thumb/images/image002.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image003.jpg",
              "fallback": Object {
                "src": "/thumb/images/image003.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image003.jpg",
                  "srcSet": "/thumb/images/image003.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image004.jpg",
              "fallback": Object {
                "src": "/thumb/images/image004.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image004.jpg",
                  "srcSet": "/thumb/images/image004.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt="005"
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image005.jpg",
              "fallback": Object {
                "src": "/thumb/images/image005.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image005.jpg",
                  "srcSet": "/thumb/images/image005.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
  </styled.div>
</React.Fragment>
`;

exports[`Gallery component custom wrapper render with custom wrapper 1`] = `
<React.Fragment>
  <styled.div
    margin={-15}
  >
    <CustomWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image001.jpg",
              "fallback": Object {
                "src": "/thumb/images/image001.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image001.jpg",
                  "srcSet": "/thumb/images/image001.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </CustomWrapper>
    <CustomWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image002.jpg",
              "fallback": Object {
                "src": "/thumb/images/image002.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image002.jpg",
                  "srcSet": "/thumb/images/image002.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </CustomWrapper>
    <CustomWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image003.jpg",
              "fallback": Object {
                "src": "/thumb/images/image003.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image003.jpg",
                  "srcSet": "/thumb/images/image003.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </CustomWrapper>
  </styled.div>
</React.Fragment>
`;

exports[`Gallery component lightboxOptions props that the Lightbox options are accepted 1`] = `
<React.Fragment>
  <styled.div
    margin={-15}
  >
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image001.jpg",
              "fallback": Object {
                "src": "/thumb/images/image001.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image001.jpg",
                  "srcSet": "/thumb/images/image001.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image002.jpg",
              "fallback": Object {
                "src": "/thumb/images/image002.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image002.jpg",
                  "srcSet": "/thumb/images/image002.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image003.jpg",
              "fallback": Object {
                "src": "/thumb/images/image003.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image003.jpg",
                  "srcSet": "/thumb/images/image003.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
  </styled.div>
</React.Fragment>
`;

exports[`Gallery component onClose prop onClose prop is accepted 1`] = `
<React.Fragment>
  <styled.div
    margin={-15}
  >
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image001.jpg",
              "fallback": Object {
                "src": "/thumb/images/image001.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image001.jpg",
                  "srcSet": "/thumb/images/image001.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image002.jpg",
              "fallback": Object {
                "src": "/thumb/images/image002.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image002.jpg",
                  "srcSet": "/thumb/images/image002.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image003.jpg",
              "fallback": Object {
                "src": "/thumb/images/image003.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image003.jpg",
                  "srcSet": "/thumb/images/image003.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
  </styled.div>
</React.Fragment>
`;

exports[`Gallery component rowMargin prop that the rowMargin prop is accepted 1`] = `
<React.Fragment>
  <styled.div
    margin={0}
  >
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image001.jpg",
              "fallback": Object {
                "src": "/thumb/images/image001.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image001.jpg",
                  "srcSet": "/thumb/images/image001.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image002.jpg",
              "fallback": Object {
                "src": "/thumb/images/image002.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image002.jpg",
                  "srcSet": "/thumb/images/image002.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
    <ImageColWrapper
      colWidth={33.333333333333336}
      gutter="0.25rem"
      mdColWidth={25}
      onClick={[Function]}
    >
      <L
        alt=""
        className=""
        image={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "images": Object {
              "alt": "/thumb/images/image003.jpg",
              "fallback": Object {
                "src": "/thumb/images/image003.jpg",
              },
              "sources": Array [
                Object {
                  "media": "/thumb/images/image003.jpg",
                  "srcSet": "/thumb/images/image003.jpg",
                  "type": "JPG",
                },
              ],
            },
            "layout": "constrained",
            "width": 100,
          }
        }
      />
    </ImageColWrapper>
  </styled.div>
</React.Fragment>
`;

exports[`Gallery component that it renders with empty props 1`] = `
<React.Fragment>
  <styled.div
    margin={-15}
  />
</React.Fragment>
`;
